src/ctap/mod.rs

```diff
@@ -534,10 +533,11 @@ where
         signature_data.extend(client_data_hash);
         let (signature, x5c) = if USE_BATCH_ATTESTATION {
```

Comment:
Maybe we can use this opportunity to remove the boolean flag and use the correct supported behavior:
If certificate exists:
  batch attestation
if not:
  self attestation

If do we think that deciding on batch attestation on compile time is a feature, and this way, you can't forget to write a certificate before usage?

Reply:
I modified the decision on batch attestation as suggested (presence of attestation) and only write the attestation to the storage if compiled with batch attestation. We can tweak later those decisions as needed (e.g. batch attestation per RP id).